File: tests/adapters/python/goldens/function_bodies/basic_strip.py
"""Module docstring."""

import os
import sys
from typing import List, Optional

class Calculator:
    """A simple calculator class."""
    
    def __init__(self, name: str = "default"):
        # … method body omitted (3 lines)
    
    def add(self, a: int, b: int) -> int:
        """Add two numbers."""
        # … method body omitted (4 lines)
    
    def multiply(self, a: int, b: int) -> int:
        # … method body omitted (5 lines)
    
    def get_history(self) -> List[str]:
        return self.history.copy() # Get calculation history

def main():
    """Main function."""
    # … function body omitted (4 lines)
    
if __name__ == "__main__":
    main()

File: tests/adapters/go/goldens/function_bodies/basic_strip.go
// Go module for testing function body optimization.
package main

import (
	"fmt"
	"strconv"
)

// User represents a user in the system
type User struct {
	ID    int
	Name  string
	Email string
}

// ProcessingResult contains valid and invalid users
type ProcessingResult struct {
	Valid   []User
	Invalid []User
}

// Calculator manages calculation history
type Calculator struct {
	name    string
	history []string
}

// NewCalculator creates a new calculator instance
func NewCalculator(name string) *Calculator // … function body omitted (10 lines)

// Add performs addition and logs the result
func (c *Calculator) Add(a, b int) int // … method body omitted (7 lines)

// Multiply performs multiplication and logs the result
func (c *Calculator) Multiply(a, b int) int // … method body omitted (6 lines)

// GetHistory returns a copy of the calculation history
func (c *Calculator) GetHistory() []string // … method body omitted (5 lines)

func validateInput(value int) bool // … function body omitted (17 lines)

// ProcessUserData validates and categorizes users
func ProcessUserData(users []User) *ProcessingResult // … function body omitted (16 lines)

func contains(s, substr string) bool // … function body omitted (8 lines)

// ProcessArray applies a processor function to each item
func ProcessArray(items []interface{}, processor func(interface{}) interface{}) []interface{} // … function body omitted (12 lines)

func main() // … function body omitted (14 lines)
